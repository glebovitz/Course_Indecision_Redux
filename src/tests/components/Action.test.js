import React from 'react';
import { shallow } from 'enzyme';
import Action from '../../components/Action';

test('should render an Action Component correctly', () => {
  const wrapper = shallow(<Action hasOptions={true}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render disabled Action Component correctly', () => {
  const wrapper = shallow(<Action hasOptions={false}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should handle button click', () => {
  const pickOption = jest.fn();
  const wrapper = shallow(<Action pickOption={pickOption} hasOptions={true} />);
  wrapper.find('button').simulate("click");
  expect(pickOption).toHaveBeenCalled();
});
