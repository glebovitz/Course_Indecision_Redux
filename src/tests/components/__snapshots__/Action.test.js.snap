// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an Action Component correctly 1`] = `
<div>
  <button
    className="big-button"
    disabled={false}
  >
    What should I do?
  </button>
</div>
`;

exports[`should render disabled Action Component correctly 1`] = `
<div>
  <button
    className="big-button"
    disabled={true}
  >
    What should I do?
  </button>
</div>
`;
